### ISSUE TITLE
Still getting auto-fetching behavior when setting is off

### ISSUE THREAD

**Issue by tristan957** [microsoft/vscode-pull-request-github#4202]
- Extension version: 0.54.1
- VSCode Version: 1.73.1
- OS: Windows 10 (Remote-SSH Fedora 36

Steps to Reproduce:

1. Have a password protected SSH key
2. Disable auto fetch for this extension
3. Consistently be prompted for SSH key password even though auto fetch is disabled

![image](https://user-images.githubusercontent.com/11936772/202233134-c8b412c6-8043-423a-baa1-fd5b08985866.png)

```jsonc
// User/settings.json
"githubRepositories.autoFetch.enabled": false,
```
This has been going on for a while.

**Comment by tristan957** [comment]
I don't even think the extension is respecting the timeout either. Holy moly is it annoying getting asked every minute to enter your password.

**Comment by alexr00** [comment]
@tristan957 thanks for reporting this! That setting doesn't belong to this extension, but I do think we need a similar setting or to just respect that setting.

**Comment by tristan957** [comment]
> @tristan957 thanks for reporting this! That setting doesn't belong to this extension, but I do think we need a similar setting or to just respect that setting.

I feel like a dummy, whoops. Thanks for your effort on this. I will just disable the extension until the next stable release, since I don't intend to switch to Insiders.

**Comment by alexr00** [comment]
> I feel like a dummy, whoops.

Not at all, this behavior was unexpected and needed to be changed.

**Comment by CombeeMike** [comment]
Do you have any concrete idea on when this will make it into production?
I'm also facing this issue and I don't want to switch to pre-release.

**Comment by alexr00** [comment]
We follow the same release schedule as VS Code: https://github.com/microsoft/vscode/issues?q=is%3Aopen+is%3Aissue+label%3Aiteration-plan+milestone%3A%22November+2022%22

**Comment by tristan957** [comment]
@CombeeMike I suggest just disabling the extension for now if that can work with your particular workflow.

**Comment by CombeeMike** [comment]
Thanks for the tip but I'm mostly in branches without an open PR ATM where this is not an issue, so I think I'm good until next week.

**Comment by alexr00** [comment]
@CombeeMike you could also try the pre-release version if you use VS Code insiders. Starting tomorrow we will be stabilizing the pre-release version for release so I wouldn't expect you to see too many issues with it.

**Comment by tristan957** [comment]
Now that I have consumed the new VSCode update in addition the latest release of this extension, everything seems a lot better.

**Comment by tristan957** [comment]
@alexr00 now that I have used the extension more with this fix, I have one more piece of feedback. Would it be possible to get the extension to turn off auto fetching in all situations? As of right now, like you said in an above comment, the extension will fetch when you checkout a branch connected to a PR. While an improvement, it is still annoying to be pestered for an SSH password when I checkout a PR branch.

For my workflow, I am usually the owner of the PR branch, so I know when it has been updated and when I need to fetch updates. In the vast majority of cases, I only ever interact with a branch on the same computer, so the latest that I have locally is always at least equal to or more recent than what is on the remote.
